packages/odata-common/src/de-serializers/de-serializers.ts

```diff
@@ -0,0 +1,87 @@
+import { identity } from '@sap-cloud-sdk/util';
+
+/**
+ * Represents a set of functions that determine (de-)serialization per EDM type.
+ */
+export interface DeSerializers<
+  BinaryT = any,
+  BooleanT = any,
+  ByteT = any,
+  DecimalT = any,
+  DoubleT = any,
+  FloatT = any,
+  Int16T = any,
+  Int32T = any,
+  Int64T = any,
+  GuidT = any,
+  SByteT = any,
+  SingleT = any,
+  StringT = any,
+  AnyT = any
+> {
+  'Edm.Binary': DeSerializer<BinaryT>;
+  'Edm.Boolean': DeSerializer<BooleanT>;
+  'Edm.Byte': DeSerializer<ByteT>;
+  'Edm.Decimal': DeSerializer<DecimalT>;
+  'Edm.Double': DeSerializer<DoubleT>;
+  'Edm.Float': DeSerializer<FloatT>;
+  'Edm.Int16': DeSerializer<Int16T>;
+  'Edm.Int32': DeSerializer<Int32T>;
+  'Edm.Int64': DeSerializer<Int64T>;
+  'Edm.Guid': DeSerializer<GuidT>;
+  'Edm.SByte': DeSerializer<SByteT>;
+  'Edm.Single': DeSerializer<SingleT>;
+  'Edm.String': DeSerializer<StringT>;
+  'Edm.Any': DeSerializer<AnyT>;
+}
+
+/**
+ * Represents a set of functions for serialization, deserialization and optionally serialization for URIs.
+ * The return type of the `deserialize` function is the type of the input to the `serialize` and `serializeToUri` functions.
+ */
+export interface DeSerializer<DeserializedT> {
+  /**
+   * Function to deserialize a value.
+   * @param val - Serialized value to deserialize.
+   * @returns Deserialized value.
+   */
+  deserialize: (val: any) => DeserializedT;
+
+  /**
+   * Function to serialize a value.
+   * @param val - Deserialized value to serialize.
+   * @returns Serialized value.
+   */
+  serialize: (val: DeserializedT) => any;
```

Comment:
Should `any` be `string`? Should we explain why or why not?

```suggestion
  /**
   * Function to serialize a value.
   * @param value - Deserialized value to serialize.
   * @returns Serialized value.
   */
  serialize: (value: DeserializedT) => any;
```